tests/sentry/mediators/sentry_app_installations/test_destroyer.py

```diff
@@ -46,6 +47,19 @@ def test_deletes_grant(self):
 
         assert not ApiGrant.objects.filter(pk=grant.id).exists()
 
+    @responses.activate
+    def test_deletes_on_read_timeout_error(self):
+        grant = self.install.api_grant
+
+        responses.add(
+            responses.POST,
+            'https://example.com/webhook',
+            body=ReadTimeout('Read timed out'))
```

Comment:
Does this work with `responses`? Like, does this result in receiving a 408 or whatever?

Reply:
@mnoble Not sure I quite understand your question. But there seems to be a difference between getting a status code that is bad (e.g. 500) and a run time exception like a `ReadTimeout` where there is no status code.

Reply:
oh I see. So our http lib would throw a `ReadTimeout` exception in this case. I guess then my question is, does `responses` do that when you set a `body` like this.

Normally, `body` is meant to be the string value of the response body. Setting it to an instance of an object is counter-intuitive to me, hence the question.

Reply:
@mnoble OK, I understand now. While it does seem weird, the docs for `responses` say you should throw errors like this: https://github.com/getsentry/responses

> Lastly, you can pass an Exception as the body to trigger an error on the request:

```
import responses
import requests

@responses.activate
def test_simple():
    responses.add(responses.GET, 'http://twitter.com/api/1/foobar',
                  body=Exception('...'))
    with pytest.raises(Exception):
        requests.get('http://twitter.com/api/1/foobar')
```

Does that answer your question?

Reply:
TIL. Yup, that does. Thanks!